chaincfg/chainec/chainec.go

```diff
@@ -38,7 +40,7 @@ type PublicKey interface {
 	GetY() *big.Int
 
 	// GetType returns the ECDSA type of this key.
-	GetType() int
+	GetType() dcrec.SignatureType
```

Comment:
The interfaces here can be moved into `dcrec/common.go`, coupled with the other changes suggested `the chainec` package can then be removed.